notion-to-md.ts

```diff
@@ -0,0 +1,223 @@
+import { Client } from "@notionhq/client";
+import { Annotations, ListBlockChildrenResponseResult, ListBlockChildrenResponseResults, ParsedMarkdownBlock, Text } from "./types";
+import * as md from "./utils/md";
+import { getBlockChildren } from "./utils/notion";
+
+interface NotionToMarkdownOptions {
+  notionClient: Client
+}
+
+export default class NotionToMarkdown {
+  notionClient: Client;
+
+  constructor(options: NotionToMarkdownOptions) {
+    this.notionClient = options.notionClient;
+  }
+
+  toMarkdownString(mdBlocks: ParsedMarkdownBlock[] = [], nestingLevel = 0) {
+    let mdString = "";
+    mdBlocks.forEach((mdBlocks) => {
+      if (mdBlocks.parent) {
+        mdString += `
+${md.addTabSpace(mdBlocks.parent, nestingLevel)}
+`;
+      }
+      if (mdBlocks.children && mdBlocks.children.length > 0) {
+        mdString += this.toMarkdownString(mdBlocks.children, nestingLevel + 1);
+      }
+    });
+    return mdString;
+  }
+
+  async pageToMarkdown(id: string) {
```

Comment:
I think this returns Array blocks, so I think we have to create an interface or maybe `notion/hq` provides something.

Reply:
yes this returns array of objects containing parent and child, notion/hq doesn't provide anything similar. So we can probably go with an interface like so:
```ts
export interface MdBlock {
  parent: string;
  children: MdBlock[] ;
}
```

Reply:
For this we probably have to model the complete data structure, from notion, and to be honest it is pretty lengthy https://developers.notion.com/reference/block

Reply:
Let's just have `any` for now.

Reply:
I already had included the type that @souvikinator suggested. It was just under a different name and wasn't explicitly specified in the function return type. I've pushed a change to use `MdBlock` as the type name. And I also updated the function documentation texts.

Regarding us using all the complete data structures as specified in the Notion API
1. We're depending on `notionhq/client` now, so all the types from notion will directly be reflected on this project. We just have to make an effort to stay up to date with the latest version of `notionhq/client`
2. The notion types don't generally concern the end-user as the goal of this package is to just return markdown strings rather than any notion types. All the types would only be of concern internally when doing the conversion.